Lib/fontTools/designspaceLib/statNames.py

```diff
@@ -214,16 +215,33 @@ def _getRibbiStyle(
     axis = axes_by_tag.get("wght")
     if axis is not None:
         for regular_label in axis.axisLabels:
-            if regular_label.linkedUserValue == userLocation[axis.name]:
+            if (
+                regular_label.linkedUserValue == userLocation[axis.name]
+                # In the "recursive" case where both the Regular has
+                # linkedUserValue pointing the Bold, and the Bold has
+                # linkedUserValue pointing to the Regular, only consider the
+                # first case: Regular (e.g. 400) has linkedUserValue pointing to
+                # Bold (e.g. 700, higher than Regular)
+                and regular_label.userValue < regular_label.linkedUserValue
+            ):
                 regularUserLocation[axis.name] = regular_label.userValue
                 bold = True
                 break
 
     axis = axes_by_tag.get("ital") or axes_by_tag.get("slnt")
     if axis is not None:
-        for urpright_label in axis.axisLabels:
-            if urpright_label.linkedUserValue == userLocation[axis.name]:
-                regularUserLocation[axis.name] = urpright_label.userValue
+        for upright_label in axis.axisLabels:
+            if (
+                upright_label.linkedUserValue == userLocation[axis.name]
+                # In the "recursive" case where both the Upright has
+                # linkedUserValue pointing the Italic, and the Italic has
+                # linkedUserValue pointing to the Upright, only consider the
+                # first case: Upright (e.g. ital=0, slant=0) has linkedUserValue
+                # pointing to Italic (e.g ital=1, slant=-12, higher than
+                # Upright in absolute value)
+                and abs(upright_label.userValue) < abs(upright_label.linkedUserValue)
```

Comment:
Because the Italic axis is 0 -> 1 (positive for italic) and the Slant axis is 0 -> -12 (negative for Italic)

Reply:
I'm trying to support both axes, as long as they follow the spec

Reply:
> Because the Italic axis is 0 -> 1 (positive for italic) and the Slant axis is 0 -> -12 (negative for Italic)

I see. Please add a comment along those lines

Reply:
If someone makes a backwards italic and then they have a positive slant value for the italic then the check might fail... maybe it would be better to check that upright_label.userValue == 0?

Reply:
There's already that comment just above

Reply:
ok sorry for the noise 😅

Reply:
What do you think about checking for upright == 0 instead?

Reply:
I'm thinking maybe it would be better actually

Reply:
> maybe it would be better to check that upright_label.userValue == 0?

I don't know. just take the greatest for ital, and the lower for slnt, whatever those are

Reply:
does it actually matter which one you pick?

Reply:
Yes actually, the spec is super clear that upright is zero, I'm going to change this:

![image](https://user-images.githubusercontent.com/3616772/196644599-9bd66cd2-f7d1-4b09-8911-48c24e19fe4c.png)

![image](https://user-images.githubusercontent.com/3616772/196644636-22b04523-208f-416c-9134-5454f0164ea8.png)

Reply:
That way the Slant axis can have the linkedUserValue either positive or negative, and both will work as expected

Reply:
well.. the spec similarly "recommends or requires" (whatever that means) that Regular wght is 400, but even that is not set in stone, is it? At least you are not that strict for wght as you are with ital/slnt in these checks above

Reply:
especially in the case of `slnt`, the spec defines it as (emphasis mine)

> "... oblique slant from _whatever the designer considers to be upright for that font design_"

so the meaning of slnt=0 being the only Upright is less clear-cut

Reply:
i think for the case of `ital` we know that 0 means upright and 1 italic, but for `slnt` we don't know whether the font is intented to lean forwards or backwards when the "Italic" button is pressed, so the use of linkedValue mapping to determine which one is the upright is more ambiguous if there's a recursive linking.

Reply:
I think the wording of both slnt and ital are similarly specific enough for this == 0 to be the "right thing" to do: both say that 0 is required to mean whatever "Regular" or Upright means for that design.

Reply:
I disagree for slnt for the reasons above and the wording "whatever the designer considers to be upright" opens up the possibility that "upright" angle not be 0

Reply:
slnt is an angle, not a simbolic boolean value, you are not supposed to write '0' if the angle is not _actually_ zero..

Reply:
I'm realizing now that the abs() check was also allowing slnt values both positive and negative... 🤦 I can revert my last commit actually.

Reply:
on re-reading the registered axes spec, it does sah "Recommended _or_ required 'Regular' value: 0 is _required_" -- so it does say 'required' indeed. I'm fine with taking whatever the spec says are the required 'Regular' values for the three registered axes for the sake of identifying the "Regular" label (which is what this function is doing). This means only wght=400 should be considered such, not anything less than..

Reply:
After thinking more, I moved back the more permissive abs() checks and I think it's right. For weight I think even more, because even though it's a bad idea, I've heard before of fonts that do style linking between e.g. Light and SemiBold or such, in addition to the usual Regular to Black. This code would allow that, and I think fontTools should not have too many opinions about what is good engineering practices, and be more on the side of allowing as much as possible, if that makes sense?

Reply:
just to be clear, this function does not control whether the linkedValue attributes are set or not in the actual STAT table, it's simply using them to figure out which ones of the labels is supposed to be the "Regular" in order to generate correct style names, in the case of recursive linking it was throwing up a RecursionError

Reply:
Exactly!